bip-0077.mediawiki

```diff
@@ -0,0 +1,285 @@
+|           subdirectory           |                  |                |
+|<---------------------------------|                  |                |
+|                                  |                  |                |
+|      destination (address,       |                  |                |
+|        subdirectory)             |                  |                |
++---------------------------------------------------->|                |
+|                                  |                  |                |
+|      Request Original PSBT       |                  |                |
++- - - - - - - - - - - - - - - - ->|                  |                |
+|                                  |  Original PSBT   |                |
+|                                  |<- - - - - - - - -|                |
+|                                  |                  |                |
+|                                  |                  |                |
+|          Original PSBT           |                  |                |
+|<---------------------------------|                  |                |
+|                                  |                  |                |
+|                                  |                  |                |
+|             Payjoin PSBT         |                  |                |
++--------------------------------->|                  |                |
+|                                  |   Payjoin PSBT   |                |
+|                                  |----------------->|                |
+|                                  |                  |                |
+|                                  |                  |   Payjoin      |
+|                                  |                  +--------------->|
+|                                  |                  |                |
++                                  +                  +                +
+
+Key: 
+|-----> Single transmission
+|- - -> Polled transmission
+</pre>
+
+===Payjoin version 2 messaging===
+
+Payjoin v2 messages use [[https://github.com/bitcoin/bips/blob/master/bip-0370.mediawiki| BIP 370 PSBT v2]] format to facilitate PSBT mutation.
+
+The payjoin version 2 protocol takes the following steps:
+
+* The recipient sends their payjoin pubkey and optional authentication credential according to the [[#enroll-messaging|enroll messaging]] protocol to receive a subdirectory allocation. It may go offline and replay enrollment to come back online.
+* Out of band, the receiver of the payment, shares a bitcoin URI with the sender including a <code>pj=</code> query parameter including the subdirectory as a base64URL encoded pubkey. To support version 1 senders the directory acts as an unsecured payjoin server so <code>pjos=0</code> must be specified in the URI. Version 2 senders may safely allow output substitution regardless. An <code>ohttp=</code> parameter containing the directory's base64URL Key Config should also be provided.
```

Comment:
```suggestion
* Out of band, the receiver of the payment shares a bitcoin URI with the sender, including a <code>pj=</code> query parameter with the subdirectory as a base64URL encoded pubkey. To support version 1 senders the directory acts as an unsecured payjoin server so <code>pjos=0</code> must be specified in the URI. Version 2 senders may safely allow output substitution regardless. An <code>ohttp=</code> parameter containing the directory's base64URL Key Config should also be provided.
```

Reply:
I rewrote the sentence but believe I fixed the core issue